pyk/src/pyk/ktool/kprove.py

```diff
@@ -81,8 +82,7 @@ class KProve(KPrint):
     def __init__(self, kompiled_directory, main_file_name=None, use_directory=None):
         super(KProve, self).__init__(kompiled_directory)
         self.directory = Path(self.kompiled_directory).parent
-        self.use_directory = (self.directory / 'kprove') if use_directory is None else Path(use_directory)
-        self.use_directory.mkdir(parents=True, exist_ok=True)
+        self.use_directory = (mkdtemp()) if use_directory is None else Path(use_directory)
```

Comment:
Something like this:>

```
     def __init__(self, kompiled_directory, main_file_name=None, use_directory=None): 
           ...
        self._use_directory = use_directory

    def use_directory(self) -> str:
        if not self._use_directory:
            self._use_directory = mkdtemp(...)
        return self._use_directory
```

I don't like the idea of IO happening in the initializer because that's not what users expect and it may cause issues like this one. It would also leave the system littered with unused temp directories otherwise.

Reply:
I don't think it shuold be a problem? @tothtamas28 is there a standard Python programming convention we can lean on here? Is it usually considered bad to have IO in the initializer?

I think `tempfile` can be set to also make sure it delete the directory on cleanup time? So we can just use that.

We should also be logging the temp file name though: `_LOGGER.info('Using temporary directory: {self.use_directory}')` or something, when we create it.

Reply:
At this point, let's just get this merged though.

Reply:
Unfortunately, `mkdtemp` doesn't do the deletion. 

> The user of [mkdtemp()](https://docs.python.org/3/library/tempfile.html#tempfile.mkdtemp) is responsible for deleting the temporary directory and its contents when done with it.

https://docs.python.org/3/library/tempfile.html#tempfile.mkdtemp

But, anyway, lets get the packaging working again.

Reply:
@F-WRunTime Acually Everett realised that [`TempoararyDirectory`](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory) works better than `mkdtemp` and automatically deletes the directory for us.
If we use that instead of `mkdtemp`, its OK to create the directory in `__init__`.

Lets keep the type of `KProve` to `Path`, so something like:

```
def __init__(...)
    ...
    if not use_directory:
        self._temp_dir = TemporaryDirectory()
        self.use_directory = self._temp_dir.name
```

Reply:
Alright sounds good. Updated and test are running on CI. Working up env setup for evm-semantics tests unless someone else has it setup we can skip waiting on me while I'm working through this exercise.